### PR TITLE
add support for epochs

### PR DESCRIPTION
**Pull request by apmellot**

### CODE REVIEW

File: meegpowreg/power_features.py [hunk 1]
```diff
@@ -67,8 +67,7 @@ def compute_features(
         fbands={'alpha': (8.0, 12.0)},
         clean_func=lambda x: x,
         n_jobs=1):
-    #doc str
-
+    """Compute features from raw data or clean epochs."""
```
Comment (agramfort) [inline meegpowreg/power_features.py:70]: you can do better. See an example in 

https://github.com/mne-tools/mne-python/blob/main/mne/event.py#L24

you could also read this page

**Review by agramfort — COMMENTED** [review]
@DavidSabbagh @dengemann can you have a look?

File: meegpowreg/power_features.py [hunk 2]
```diff
@@ -17,6 +19,15 @@ def _compute_covs(raw, clean_events, fbands, duration):
     return np.array(covs)
 
 
+def _compute_covs_epochs(epochs, fbands):
+    covs = list()
+    for _, fb in fbands.items():
+        ec = epochs.copy().load_data().filter(fb[0], fb[1])
+        cov = mne.compute_covariance(ec, method='oas', rank=None)
```
Comment (dengemann) [inline meegpowreg/power_features.py:26]: OK for now to have this hard-coded. Later these parameters should be exposed.

File: meegpowreg/tests/test_power_features.py [hunk 3]
```diff
@@ -31,3 +31,33 @@ def test_compute_features():
     assert features['xfreqcorrs'].shape == (n_fb * n_channels,
                                             n_fb * n_channels)
     assert features['cospcovs'].shape[1:] == (n_channels, n_channels)
+
+
+def test_compute_features_epochs():
+    raw = mne.io.read_raw_fif(raw_fname, verbose=False)
+    raw = raw.copy().crop(0, 200).pick(
+        list(range(2)) + list(range(330, 333))  # take some MEG and EEG
```
Comment (dengemann) [inline meegpowreg/tests/test_power_features.py:39]: wouldn't it be more readable to just type the random picks: ```[0, 1, 330, 331, 332]``` ?

File: meegpowreg/tests/test_power_features.py [hunk 4]
```diff
@@ -31,3 +31,33 @@ def test_compute_features():
...
+    epochs = mne.Epochs(raw, events, event_id=3000, tmin=0, tmax=60.,
+                        proj=True, baseline=None, reject=None,
+                        preload=True, decim=1)
+    features, res = compute_features(epochs, fbands=fbands)
+    n_channels = len(raw.ch_names)
+    n_freqs = len(features['freqs'])
+    n_fb = len(fbands)
+    assert (
+        set(features.keys()) ==
+        set(['psds', 'freqs', 'covs', 'xfreqcovs', 'xfreqcorrs', 'cospcovs'])
```
Comment (dengemann) [inline meegpowreg/tests/test_power_features.py:55]: you could use set literals here `{'psds', 'freqs', ...}`

**Comment by dengemann** [comment]
@apmellot thank you for the PR! I have only minor comments. It looks as if this code would allow us to reproduce the pipeline from our papers. If you address the 2 mini issues I have raised OK to merge from my side.

**Comment by DavidSabbagh** [comment]
Thank you for the PR Apolline. I have comments I will share with you tomorrow morning.

**Comment by DavidSabbagh** [comment]
@dengemann you merged this PR but it seems that my comments have not been taken into account. Most of them were cosmetic but my first 2 comments were important I think

**Comment by dengemann** [comment]
I could not find your comments for some reason.
And I had to move on.
If you care about it please do PR or push directly to master if it's minor.

**Comment by agramfort** [comment]
@DavidSabbagh you commented but I am sure you forgot to submit the review so we don't see them

**Comment by DavidSabbagh** [comment]
you're right Alex, sorry

**Pull request merged**